plugins/chromatic_correction.py

```diff
@@ -0,0 +1,252 @@
+                    im.metadata.update({model.MD_ROTATION_COR: rotation})
+                    im.metadata.update({model.MD_SHEAR_COR: shear})
+
+                    # save the image and display
+                    self.show_static_stream(f"modified_{im.metadata[model.MD_DESCRIPTION]}_{i}.ome.tiff", im)
+                except (ValueError, KeyError) as ex:
+                    logging.error("Failed to compute chromatic aberration correction yaml file due to: %s", ex)
+
+        return trans_per_channel
+
+    def start(self):
+        localization_tab = self.main_app.main_data.getTabByName("cryosecom-localization")
+        tab_data = localization_tab.tab_data_model
+        other_mip_streams = []
+        reference_mip_stream = []
+
+        if len(tab_data.streams.value) <= 1:
+            box = wx.MessageDialog(self.main_app.main_frame,
+                                   "Add minimum two streams. One of the streams must be green channel.",
+                                   "Failed to do chromatic correction", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        levels = generate_zlevels(tab_data.main.focus,
+                                  (-10e-06, 10e-06),
+                                  5e-06)
+        non_static = [s for s in tab_data.streams.value
+                      if isinstance(s, FluoStream)]
+        zlevels = {s: levels for s in non_static}
+        settings_observer = tab_data.main.settings_obs
+        acqui_task = acquireZStack(non_static, zlevels, settings_obs=settings_observer)
+        das, e = acqui_task.result()
+
+        # Calculate Maximum Intensity projection for all the present channels in the stream panel
+        for i, da in enumerate(das):
+            mip_image = model.DataArray(numpy.amax(da, axis=0), copy.copy(da.metadata))
+            dataio.tiff.export(f"{da.metadata[model.MD_DESCRIPTION]}_{i}.ome.tiff", mip_image)
+            # All the channels will be corrected according to a reference colored channel
+            # green channel is selected as reference channel, as it lies in the middle of the spectrum
+            if da.metadata[model.MD_USER_TINT] == (0, 255, 0):
+                reference_mip_stream = mip_image
+            else:
+                other_mip_streams.append(mip_image)
+
+        # Save and display the reference channel in the analysis tab
+        self.show_static_stream(f"modified_{reference_mip_stream.metadata[model.MD_DESCRIPTION]}_ref.ome.tiff",
+                                reference_mip_stream)
+
+        # Compute transformation between reference channel and other channels
+        max_distance = ACTUAL_BEAD_DIAMETER * SCALE  # size of bead in the image
+        trans_per_channel = self.compute_transformation(reference_mip_stream, other_mip_streams, max_distance)
+
+        if trans_per_channel:
+            file_location = os.path.join(odemis.gui.conf.file.CONF_PATH, "lens-chromatic-correction.odm.yaml")
```

Comment:
I don't think it's a really convenient place, because it's in a hidden ".config" folder. (and that's not the folder where the microscope file is stored). I think it's nicer for the user to store it in `~/Documents/`.